Plugin.php

```diff
@@ -49,38 +75,51 @@ class Plugin implements
 	];
 
 	/**
-	 * @var array<string, array{hook_type: string, types: list<Union>}>
+	 * @var array<string, Hook>
 	 */
 	public static $hooks = [];
 
+	/**
+	 * @var string[]
+	 */
+	public static $parseErrors = [];
+
 	public function __invoke( RegistrationInterface $registration, ?SimpleXMLElement $config = null ) : void {
 		$registration->registerHooksFromClass( static::class );
 
-		// If useDefaultStubs is not set or set to anything except false,
-		// we want to load the stubs included in this plugin.
+		if ( isset( $config->requireAllParams['value'] ) && $config->requireAllParams['value'] === 'true' ) {
+			static::$requireAllParams = true;
+		}
```

Comment:
For consistency, there should be a comment for `requireAllParams`, similar to the ones for the other options.
Furthermore, the `requireAllParams` option should be documented in the `README.md`.